### ISSUE TITLE
archinstall.lib.exceptions.SysCallError: ... unknown filesystem type 'crypto_LUKS'

### ISSUE THREAD

**Issue by krotera** [archlinux/archinstall#209]
After pressing Enter to format the disk,
https://i.imgur.com/trJOmbI.png

**Comment by dylanmtaylor** [comment]
This is with existing partitions on the disk? GPT or MBR? Are you trying to wipe the whole disk? 

Perhaps try this as a workaround (completely wipes the disk):
```
gdisk /dev/sda
x
z
y
y
```

Then reboot and try again.

**Comment by Torxed** [comment]
This was an issue we had in a older version of Archinstall.
I'm surprised to see it still. But we'll make this more robust for the next version.

It's essentially an issue where old partitions trick causes the safety mechanisms to go haywire.

If you're OK with wiping the entire disk:
-----------------------------
As Dylan mentioned, a workaround is to manually wipe the disk once before running the installer, either with `gdisk` or simply:
```
parted -s /dev/sda mklabel msdos
```
This should clear any partition tables.

**Comment by dylanmtaylor** [comment]
I think that #124 will help address partitioning issues

**Comment by krotera** [comment]
> This is with existing partitions on the disk? GPT or MBR? Are you trying to wipe the whole disk?

GPT drive with a pre-existing Linux install.

I was trying to reformat the entire drive with a new Arch install with disk encryption.

> Perhaps try this as a workaround (completely wipes the disk):
> 
> ```
> gdisk /dev/sda
> x
> z
> y
> y
> ```
> 
> Then reboot and try again.

> It's essentially an issue where old partitions trick causes the safety mechanisms to go haywire.
> 
> As Dylan mentioned, a workaround is to manually wipe the disk once before running the installer

This worked! :partying_face:

After finding a different drive to test with, I tried repeating the process. Same situation: GPT drive with a pre-existing Linux installation, which I tried reformatting completely with a new Arch system via the installer. 

This time, I also got an error, but a different one: [`archinstall.lib.exception.SysCallError: No key available with this passphrase`](https://i.imgur.com/BbU8NiV.png)

Wiping the drive via
```
gdisk /dev/sda
x
z
y
y
```
then rebooting and trying again with the same installer choices worked.

**Comment by Torxed** [comment]
I'm glad that worked, although an annoying step.
I only got one followup questions so I can re-create this issue:
 * Which pre-existing Linux install existed and did it use disk encryption? It would help a lot to know the disk layout and partition types to re-create the issue.
 
*(`lsblk -f -o+TYPE,SIZE` before the installation and `lsblk -f -o+TYPE,SIZE` after the installation would be awesome to have if you're ever re-installing again)*

**Comment by krotera** [comment]
@Torxed For what it's worth, from the foggy halls of memory:
- The first drive I tried (with the original error) had a Manjaro install with disk encryption. It had a LUKS-encrypted root partition, a (I *think* also encrypted) swap partition, and an EFI boot partition.
- The second drive I tried (which was wiped, and then it worked) had an Arch install with NO disk encryption. Its partition layout was the same (a root partition, a swap partition, and an EFI boot partititon).

**Comment by Torxed** [comment]
> @Torxed For what it's worth, from the foggy halls of memory:
> 
> * The first drive I tried (with the original error) had a Manjaro install with disk encryption. It had a LUKS-encrypted root partition, a (I _think_ also encrypted) swap partition, and an EFI boot partition.
> * The second drive I tried (which was wiped, and then it worked) had an Arch install with NO disk encryption. Its partition layout was the same (a root partition, a swap partition, and an EFI boot partititon).

Awesome, thank you! I'll install Manjaro and then try to run it over :bullettrain_front: and see if I can recreate this issue.

**Comment by MrTastyBiscuits** [comment]
> After pressing Enter to format the disk,
> https://i.imgur.com/trJOmbI.png

I have the exact same issue (and error message) on a 2TB NVMe SSD that previously contained a LUKS encrypted Manjaro.
I have tried to format it using either `gdisk`or `parted` and rebooting afterwards. However, every time I try to use `archinstall` again, it fails with the same error message as in krotera's picture. 

I also tried to format the disk to simple unencrypted ext4 and run `archinstall` without formatting it before. This time I got the error 

I must be doing something wrong but I can't figure out what…

**Comment by Torxed** [comment]
> > After pressing Enter to format the disk,
> > https://i.imgur.com/trJOmbI.png
> 
> I have the exact same issue (and error message) on a 2TB NVMe SSD that previously contained a LUKS encrypted Manjaro.
> I have tried to format it using either `gdisk`or `parted` and rebooting afterwards. However, every time I try to use `archinstall` again, it fails with the same error message as in krotera's picture.
> 
> I also tried to format the disk to simple unencrypted ext4 and run `archinstall` without formatting it before. This time I another error, but it still won't mount.
> edit: the error reads something along those lines:
> 
> ```
> archinstall.lib.exception.SysCallError: mount /tmp/xyz…xyz: wrong fs type, bad option, bad superblock on /dev/nvme1n1p1, missing codepage or helper program, or other error.
> 
> mount … did not exit gracefully…
> ```
> 
> I must be doing something wrong but I can't figure out what…

Hopefully the new release that came today sorts this issue. But if you wouldn't mind, could you give us the output of `lsblk -o+PARTUUID,UUID,TYPE,SIZE` before and after running archinstall? :)

**Comment by MrTastyBiscuits** [comment]
Thanks for you swift reply.
Here is a picture of the command that I ran before (on the top) and after (bottom). I hope that it is legible : https://imgur.com/a/9CEWlOc
I have been trying to install Arch on nvme1n1.

**Comment by Absinthe31** [comment]
> > After pressing Enter to format the disk,
> > https://i.imgur.com/trJOmbI.png
> 
> I have the exact same issue (and error message) on a 2TB NVMe SSD that previously contained a LUKS encrypted Manjaro.
> I have tried to format it using either `gdisk`or `parted` and rebooting afterwards. However, every time I try to use `archinstall` again, it fails with the same error message as in krotera's picture.
> 
> I also tried to format the disk to simple unencrypted ext4 and run `archinstall` without formatting it before. This time I another error, but it still won't mount.
> edit: the error reads something along those lines:
> 
> ```
> archinstall.lib.exception.SysCallError: mount /tmp/xyz…xyz: wrong fs type, bad option, bad superblock on /dev/nvme1n1p1, missing codepage or helper program, or other error.
> 
> mount … did not exit gracefully…
> ```
> 
> I must be doing something wrong but I can't figure out what…

Same issue here

**Comment by Torxed** [comment]
@here I assume that these issues are all related to the official ISO which has the old version (We released a new version yesterday that *should* address this issue).

If anyone is willing to test the latest release, and make sure this issue is squashed, feel free to download a pre-built updated ISO here: https://github.com/archlinux/archinstall/releases/tag/v2.1.4-RC2 which contains v2.1.4.
I'll close this until confirmation that it isn't only the old release that has this issue.

(I appreciate the reports and pictures tho!)

**Comment by MrTastyBiscuits** [comment]
> @here I assume that these issues are all related to the official ISO which has the old version (We released a new version yesterday that _should_ address this issue).
> 
> If anyone is willing to test the latest release, and make sure this issue is squashed, feel free to download a pre-built updated ISO here: https://github.com/archlinux/archinstall/releases/tag/v2.1.4-RC2 which contains v2.1.4.
> I'll close this until confirmation that it isn't only the old release that has this issue.
> 
> (I appreciate the reports and pictures tho!)

TL;DR It worked but not on my first try. Thanks !

I gave it two tries with the RC2 ISO:

**1.**
On the first one (with a freshly formatted drive), the following error still occurs and is displayed:

> ```
> archinstall.lib.exception.SysCallError: mount /tmp/xyz…xyz: wrong fs type, bad option, bad superblock on /dev/nvme1n1p1, missing codepage or helper program, or other error.
>
> mount … did not exit gracefully…
> ```


**However**, it did not prevent `archinstall` from continuing with the installation and it seems to be installing the system correctly. 
But, upon rebooting, I was not able to unlock the LUKS partition (`No key available with this passphrase`) with the password I set. So, either I am an idiot or the mount issue has something to do with that (could be both haha).

**2.**
I used `archinstall` a second time without formatting the drive with the aim to overwrite what I just did with the exact same parameters.
This time, no error whatsoever, Arch was installed and is now working perfectly.

**Comment by bf** [comment]
Had same issue when running archinstall on my nvme. Doing a `parted -s /dev/nvme0n1 mklabel msdos` before running archinstall solved the issue.

**Comment by Torxed** [comment]
> Had same issue when running archinstall on my nvme. Doing a `parted -s /dev/nvme0n1 mklabel msdos` before running archinstall solved the problem.

Even with the latest version of archinstall?
I believed this was fixed a long time ago.

**Comment by bf** [comment]
Yes, with latest version. TBH I fucked up the partitions on the drive before as well. It's an NVMe which needs the `nvme_core.default_ps_max_latency_us=0` kernel parameter
